DataFormats/PortableTestObjects/src/classes_def.xml

```diff
@@ -3,16 +3,16 @@
   <class name="portabletest::TestSoA::View"/>
 
   <class name="portabletest::TestHostCollection"/>
-  <read
+  <!--read
     sourceClass="portabletest::TestHostCollection"
     targetClass="portabletest::TestHostCollection"
     version="[1-]"
     source="portabletest::TestSoA layout_;"
-    target="buffer_,layout_,view_"
+    target=""
```

Comment:
I think I misunderstood how the `target` directive works.
After reading the code generated by ROOT, I don't think we need to specify these, even if we were to stay with the XML approach.

Reply:
I though the content of `target` were needed to avoid memory leaks, and tell ROOT this read rule needs to be run if any of those member variables are asked for (https://github.com/cms-sw/cmssw/issues/40491, https://github.com/cms-sw/cmssw/pull/40492).

Reply:
They definitively are needed.  They may not change the content of the generated function but they affect significantly the way TStreamerInfo handles the rules.  As Matti said, not specifying will lead to 'huge' memory leak.

Reply:
Can you show me _what_ is affected by the definition of the `target` entry ?

Reply:
It is a bit complicated and indirect but you can start by looking the consequence of those lines:
```
io/io/src/TStreamerInfo.cxx
639:             && rules && !rules.HasRuleWithTarget( element->GetName(), kTRUE ) )
2508:                && !rules.HasRuleWithTarget( element->GetName(), kTRUE ) ) {
2530:         } else if (rules.HasRuleWithTarget( element->GetName(), kTRUE ) ) {
2542:      } else if (rules && rules.HasRuleWithTarget( element->GetName(), kTRUE ) ) {

io/io/src/TStreamerInfo.cxx
4548:         if ( rule->HasTarget( element->GetName() ) ) {

io/io/src/TStreamerInfo.cxx
1850:               const TObjArray* targets = rule->GetTarget();
4595:      if (rule->GetTarget()==0) {
4607:         toAdd.reserve(rule->GetTarget()->GetEntriesFast());
4608:         TObjString * objstr = (TObjString*)(rule->GetTarget()->At(0));
4625:            for(Int_t other = 1; other < rule->GetTarget()->GetEntriesFast(); ++other) {
4626:               objstr = (TObjString*)(rule->GetTarget()->At(other));
```
In particular line 639, 2508, 4548, 4626 or even 4595

Reply:
Thanks.

So the effect is not in the code generated by the read rules, but deeper in `TStreamerInfo`.

Reply:
Yes, exactly.